### DISCUSSION TITLE
Updating Fabric

### DISCUSSION THREAD

**Discussion by secretgspot** [danielmiessler/Fabric#578] [Q&A]
what are steps to update? do I just git pull?

**Comment by tanng1999** [comment]
I do the following steps to get new version of Fabric.

1. Find out the current version of Fabric on local system
```bash
pipx list
```
2. Git pull to get the latest changes from Github
```bash
cd fabric
git pull
```
3. Update Fabric using pipx
```bash
pipx upgrade fabric
```
4. Find out the current version of Fabric on local system again
```bash
pipx list
```

My current system shows I am on `fabric 1.2.0`.  However, on the [releases page](https://github.com/danielmiessler/fabric/releases), it is shows the latest version is `fabric 1.4.0`.  I think the [project.toml](https://github.com/danielmiessler/fabric/blob/main/pyproject.toml) file has not been updated to reflect the new `1.4.0` version yet.
